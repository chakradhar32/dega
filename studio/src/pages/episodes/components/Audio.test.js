import React from 'react';
import { Provider } from 'react-redux';
import { act } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';

import '../../../matchMedia.mock';
import Audio from './Audio';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('@editorjs/editorjs');
jest.mock('../../../components/Uppy', () => ({
  UppyUploader: jest.fn().mockReturnValue([{ url: { raw: 'returnedUrl' } }]),
}));

describe('Audio Component', () => {
  describe('snapshot testing', () => {
    it('should render the component', () => {
      const onUpload = jest.fn();
      const tree = mount(<Audio onUpload={onUpload} />);
      expect(tree).toMatchSnapshot();
    });
    it('should render the component with url', () => {
      const onUpload = jest.fn();
      const url = 'dummyUrl';
      const tree = mount(<Audio onUpload={onUpload} url={url} />);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    it('should remove audio on click of remove button', () => {
      const onUpload = jest.fn();
      const url = 'dummyUrl';
      const store = mockStore({
        spaces: {
          orgs: [{ id: 1, organisation: 'Organisation 1', spaces: [11] }],
          details: {
            11: { id: 11, name: 'Space 11' },
          },
          loading: false,
          selected: 11,
        },
      });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <Audio onUpload={onUpload} url={url} />
          </Provider>,
        );
      });
      const selectButton = wrapper.find('Button').at(0);

      expect(selectButton.text()).toBe('Remove');
      selectButton.simulate('click');
    });
  });
});
