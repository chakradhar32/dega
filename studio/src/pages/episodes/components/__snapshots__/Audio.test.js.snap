// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Component snapshot testing should render the component 1`] = `
<Audio
  onUpload={[MockFunction]}
>
  <Modal
    closable={false}
    confirmLoading={false}
    footer={
      Array [
        <ForwardRef(InternalButton)
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          Return
        </ForwardRef(InternalButton)>,
        <ForwardRef(InternalButton)
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Ok
        </ForwardRef(InternalButton)>,
      ]
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={false}
    width="800px"
  >
    <Dialog
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <ForwardRef(CloseOutlined)
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        Array [
          <ForwardRef(InternalButton)
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
          >
            Return
          </ForwardRef(InternalButton)>,
          <ForwardRef(InternalButton)
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            type="primary"
          >
            Ok
          </ForwardRef(InternalButton)>,
        ]
      }
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={false}
      width="800px"
      wrapClassName=""
    >
      <PortalWrapper
        visible={false}
      />
    </Dialog>
  </Modal>
  <Space
    direction="vertical"
  >
    <div
      className="ant-space ant-space-vertical"
    >
      <Item
        className="ant-space-item"
        direction="vertical"
        index={0}
        key="ant-space-item-0"
        marginDirection="marginRight"
        size="small"
      />
      <Item
        className="ant-space-item"
        direction="vertical"
        index={1}
        key="ant-space-item-1"
        marginDirection="marginRight"
        size="small"
      >
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <Space
            direction="horizontal"
          >
            <div
              className="ant-space ant-space-horizontal ant-space-align-center"
            >
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                size="small"
              />
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                size="small"
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }
                >
                   
                </div>
              </Item>
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={2}
                key="ant-space-item-2"
                marginDirection="marginRight"
                size="small"
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <Button
                    block={false}
                    ghost={false}
                    htmlType="button"
                    loading={false}
                    onClick={[Function]}
                  >
                    <Wave>
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        <LoadingIcon
                          existIcon={false}
                          loading={false}
                          prefixCls="ant-btn"
                        >
                          <CSSMotion
                            motionName="ant-btn-loading-icon-motion"
                            onAppearActive={[Function]}
                            onAppearStart={[Function]}
                            onEnterActive={[Function]}
                            onEnterStart={[Function]}
                            onLeaveActive={[Function]}
                            onLeaveStart={[Function]}
                            removeOnLeave={true}
                            visible={false}
                          >
                            <DomWrapper />
                          </CSSMotion>
                        </LoadingIcon>
                        <span
                          key=".0"
                        >
                          Select
                        </span>
                      </button>
                    </Wave>
                  </Button>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </Item>
    </div>
  </Space>
</Audio>
`;

exports[`Audio Component snapshot testing should render the component with url 1`] = `
<Audio
  onUpload={[MockFunction]}
  url="dummyUrl"
>
  <Modal
    closable={false}
    confirmLoading={false}
    footer={
      Array [
        <ForwardRef(InternalButton)
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          Return
        </ForwardRef(InternalButton)>,
        <ForwardRef(InternalButton)
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Ok
        </ForwardRef(InternalButton)>,
      ]
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={false}
    width="800px"
  >
    <Dialog
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <ForwardRef(CloseOutlined)
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        Array [
          <ForwardRef(InternalButton)
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
          >
            Return
          </ForwardRef(InternalButton)>,
          <ForwardRef(InternalButton)
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            type="primary"
          >
            Ok
          </ForwardRef(InternalButton)>,
        ]
      }
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={false}
      width="800px"
      wrapClassName=""
    >
      <PortalWrapper
        visible={false}
      />
    </Dialog>
  </Modal>
  <Space
    direction="vertical"
  >
    <div
      className="ant-space ant-space-vertical"
    >
      <Item
        className="ant-space-item"
        direction="vertical"
        index={0}
        key="ant-space-item-0"
        marginDirection="marginRight"
        size="small"
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          <audio
            controls={true}
          >
            <source
              src="dummyUrl"
              type="audio/ogg"
            />
          </audio>
        </div>
      </Item>
      <Item
        className="ant-space-item"
        direction="vertical"
        index={1}
        key="ant-space-item-1"
        marginDirection="marginRight"
        size="small"
      >
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <Space
            direction="horizontal"
          >
            <div
              className="ant-space ant-space-horizontal ant-space-align-center"
            >
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                size="small"
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }
                >
                  <Button
                    block={false}
                    ghost={false}
                    htmlType="button"
                    loading={false}
                    onClick={[Function]}
                  >
                    <Wave>
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        <LoadingIcon
                          existIcon={false}
                          loading={false}
                          prefixCls="ant-btn"
                        >
                          <CSSMotion
                            motionName="ant-btn-loading-icon-motion"
                            onAppearActive={[Function]}
                            onAppearStart={[Function]}
                            onEnterActive={[Function]}
                            onEnterStart={[Function]}
                            onLeaveActive={[Function]}
                            onLeaveStart={[Function]}
                            removeOnLeave={true}
                            visible={false}
                          >
                            <DomWrapper />
                          </CSSMotion>
                        </LoadingIcon>
                        <span
                          key=".0"
                        >
                          Remove
                        </span>
                      </button>
                    </Wave>
                  </Button>
                </div>
              </Item>
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                size="small"
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }
                >
                   
                </div>
              </Item>
              <Item
                className="ant-space-item"
                direction="horizontal"
                index={2}
                key="ant-space-item-2"
                marginDirection="marginRight"
                size="small"
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <Button
                    block={false}
                    ghost={false}
                    htmlType="button"
                    loading={false}
                    onClick={[Function]}
                  >
                    <Wave>
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        <LoadingIcon
                          existIcon={false}
                          loading={false}
                          prefixCls="ant-btn"
                        >
                          <CSSMotion
                            motionName="ant-btn-loading-icon-motion"
                            onAppearActive={[Function]}
                            onAppearStart={[Function]}
                            onEnterActive={[Function]}
                            onEnterStart={[Function]}
                            onLeaveActive={[Function]}
                            onLeaveStart={[Function]}
                            removeOnLeave={true}
                            visible={false}
                          >
                            <DomWrapper />
                          </CSSMotion>
                        </LoadingIcon>
                        <span
                          key=".0"
                        >
                          Select
                        </span>
                      </button>
                    </Wave>
                  </Button>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </Item>
    </div>
  </Space>
</Audio>
`;
